src/app/components/dialog/dialog.js

```diff
@@ -71,3 +72,36 @@ export default function Dialog({
         </ReactModal>
     );
 }
+
+export function useDialog() {
+    const [showDialog, updateShowDialog] = React.useState(false);
+    const open = () => updateShowDialog(true);
+    const close = () => updateShowDialog(false);
+
+    function BoundDialog({
+        title, children, modal = true, className,
+        showPutAway=true,
+        ...otherProps
+    }) {
+        const Modal = modal ? ReactModal : React.Fragment;
+
+        return (
+            <Modal
+                isOpen={showDialog}
+                className='modal'
+                overlayClassName="modal-overlay"
+                bodyOpenClassName="no-scroll-dialog"
+                {...otherProps}
+            >
+                <Dialog
+                    title={title} className={className} isOpen={showDialog}
+                    onPutAway={showPutAway && close}
+                >
+                    {children}
+                </Dialog>
+            </Modal>
+        );
+    }
```

Comment:
you'll get a new instance of `BoundDialog`, open & close on every render, this will force react to unmount & remount on every render since the component will change. If the returned `BoundDialog` didn't change it would just re-render and probably not need to change any DOM.  I think you could use useMemo here to implement that

Reply:
I will revisit this.